### PR TITLE
refactor: replace archived action

### LINKED ISSUE #571: CI has `54 warnings` under `Annotations`

### PR DESCRIPTION
**Pull request by hamirmahal**
fixes #571.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -36,12 +36,7 @@ jobs:
 
       - name: install ${{ matrix.toolchain }} toolchain
         id: install_toolchain
-        uses: actions-rs/toolchain@v1
```
Comment (hamirmahal) [inline .github/workflows/ci.yml:39]: That repository is archived, and it doesn't seem like it's going to receive any updates anytime soon.

https://github.com/actions-rs/toolchain/pull/220

**Comment by hamirmahal** [comment]
# Without this change

image

# With this change

image

**Comment by hamirmahal** [comment]
# Without this change
https://github.com/RustAudio/rodio/actions/runs/8711480431

image

# With this change
https://github.com/RustAudio/rodio/actions/runs/8763646895?pr=572

image

**Review by dvdsk — APPROVED** [review]

**Comment by dvdsk** [comment]
Nice, now CI is without any warnings :)

**Pull request merged**